File: 111-MinimumDepthofBinaryTree/111-MinimumDepthofBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if root == None: return 0
        
        if root.left == None and root.right == None:
            return 1
        
        left = self.minDepth(root.left) if root.left else float('inf')
        right = self.minDepth(root.right) if root.right else float('inf')
    
        
        return min(left, right) + 1
        

File: 111-MinimumDepthofBinaryTree/111-MinimumDepthofBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    // case if none
    if(root == null) return 0;

    // case if no children
    if(root.left == null && root.right == null) return 1;

    // case if one children or the other
    let left = root.left ? minDepth(root.left) : Infinity
    let right = root.right ? minDepth(root.right) : Infinity
    return Math.min(left, right) + 1;
};
